src/games/strategy/engine/lobby/client/ui/LobbyGameTableModel.java

```diff
@@ -54,47 +60,58 @@ public void gameRemoved(final GUID gameId) {
         assertSentFromServer();
         removeGame(gameId);
       }
-    }, ILobbyGameBroadcaster.GAME_BROADCASTER_CHANNEL);
+    };
+    m_channelMessenger.registerChannelSubscriber(lobbyGameBroadcaster, ILobbyGameBroadcaster.GAME_BROADCASTER_CHANNEL);
+
     final Map<GUID, GameDescription> games =
         ((ILobbyGameController) m_remoteMessenger.getRemote(ILobbyGameController.GAME_CONTROLLER_REMOTE)).listGames();
     for (final GUID id : games.keySet()) {
-      addGame(id, games.get(id));
-    }
-  }
-
-  public GameDescription get(final int i) {
-    return m_games.get(i);
-  }
-
-  @Override
-  public Class<?> getColumnClass(final int columnIndex) {
-    if (columnIndex == getColumnIndex(Column.Started)) {
-      return Date.class;
+      updateGame(id, games.get(id));
     }
-    return Object.class;
   }
 
   private void removeGame(final GUID gameId) {
     SwingUtilities.invokeLater(new Runnable() {
       @Override
       public void run() {
-        final int index = m_gameIDs.indexOf(gameId);
-        m_gameIDs.remove(index);
-        m_games.remove(index);
-        fireTableRowsDeleted(index, index);
+        if (gameId == null) {
+          return;
```

Comment:
For the new null checks on bad input, is it better to return silently or throw an exception?